test/e2e/specs/site-editor/new-templates-list.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Templates', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'emptytheme' ),
+			requestUtils.activatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'twentytwentyone' ),
+			requestUtils.deactivatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test( 'Sorting', async ( { admin, page } ) => {
+		await admin.visitSiteEditor( { path: '/wp_template/all' } );
+		// Descending by title.
+		await page.getByRole( 'button', { name: 'Template' } ).click();
```

Comment:
It's a bit unclear to me at first glance that we're clicking on the column header. Perhaps this is clearer?

```js
const table = page.getByRole( 'region', { name: 'Templates' } )
  .getByRole( 'table' );

await table
  .getByRole( 'columnheader', { name: 'Template' } )
  .getByRole( 'button', { name: 'Template' } )
  .click();
```